_posts/2020-09-15-mutiny-pagination.adoc

```diff
@@ -0,0 +1,200 @@
+---
+layout: post
+title: 'Handling paginated APIs with Mutiny'
+date: 2020-09-15
+tags: reactive rest mutiny http pagination
+synopsis: How to use API using pagination with Mutiny
+author: cescoffier
+---
+
+At the beginning of the Mutiny adventure, my friend Alex came to me with an interesting problem. 
+Alex wanted to retrieve data from a REST service in a reactive manner.
+So far, no problem, we have everything for this in our toolbox. 
+But, this service, as many services, is using pagination. 
+Ah! That makes things a bit more spicy.
+Alex wanted to retrieve all the items and consumes them as a stream, but you can't retrieve the items in one batch.
+You need to invoke the service for every page, extract the items and feed the stream.
+
+So, how to achieve this in a reactive manner and build a proper stream of items without loosing your sanity?
+Let's have a look!
+
+## The Punk API
+
+First, we need an API.
+Alex introduced me to the https://punkapi.com/documentation/v2[Punk API], a REST API to retrieve beers.
+That's fun, and even better, it uses pagination. 
+We got our API!
+
+If you call `https://api.punkapi.com/v2/beers?page=1`, you get a JSON array like:
+
+[source, json]
+----
+[
+    {
+        first beer
+    },
+    {
+        second beer
+    }, 
+    // ...
+]
+----
+
+I won't details the content of each object, the documentation page does a great job about that.
```

Comment:
Maybe missing "show" or something like that between "won't" and "details?

Reply:
Fixed.